src/visualization/utilities.jl

```diff
@@ -85,33 +85,142 @@ function global_plotting_triangulation_triplot(xyz_plot, u_plot, t)
+
+  # extract node coordinates
+  uEltype = eltype(first(u))
+  nvars = nvariables(equations)
+  n_nodes_2d = nnodes(dg)^ndims(mesh)
+  n_elements = nelements(dg, cache)
+  x = reshape(view(cache.elements.node_coordinates, 1, :, :, :), n_nodes_2d, n_elements)
+  y = reshape(view(cache.elements.node_coordinates, 2, :, :, :), n_nodes_2d, n_elements)
+
+  # extract indices of local face nodes for wireframe plotting
+  Fmask = get_face_node_indices(r, s, dg)
+  plotting_interp_matrix1D = face_plotting_interpolation_matrix(dg; nvisnodes=nvisnodes)
+
+  # These 5 lines extract the face values on each element from the arrays x,y,sol_to_plot.
+  # The resulting arrays are then reshaped so that xf, yf, sol_f are Matrix types of size
+  # (Number of face plotting nodes) x (Number of faces).
+  function face_first_reshape(x, num_nodes_1D, num_nodes, num_elements)
+      num_reference_faces = 2 * ndims(mesh)
+      xf = view(reshape(x, num_nodes, num_elements), vec(Fmask), :)
+      return reshape(xf, num_nodes_1D, num_elements * num_reference_faces)
+  end
+  reshape_and_interpolate(x) = plotting_interp_matrix1D * face_first_reshape(x, nnodes(dg), n_nodes_2d, n_elements)
+  xfp, yfp = map(reshape_and_interpolate, (x, y))
+  ufp = StructArray{SVector{nvars, uEltype}}(map(reshape_and_interpolate, StructArrays.components(u)))
+
+  return xfp, yfp, ufp
+end
+
+function mesh_plotting_wireframe(u::ScalarData, mesh, equations, dg::DGSEM, cache; nvisnodes=2*nnodes(dg))
+
+  # build nodes on reference element (seems to be the right ordering)
+  r, s = reference_node_coordinates_2d(dg)
+
+  # extract node coordinates
+  n_nodes_2d = nnodes(dg)^ndims(mesh)
+  n_elements = nelements(dg, cache)
+  x = reshape(view(cache.elements.node_coordinates, 1, :, :, :), n_nodes_2d, n_elements)
+  y = reshape(view(cache.elements.node_coordinates, 2, :, :, :), n_nodes_2d, n_elements)
+
+  # extract indices of local face nodes for wireframe plotting
+  Fmask = get_face_node_indices(r, s, dg)
+  plotting_interp_matrix1D = face_plotting_interpolation_matrix(dg; nvisnodes=nvisnodes)
+
+  # These 5 lines extract the face values on each element from the arrays x,y,sol_to_plot.
+  # The resulting arrays are then reshaped so that xf, yf, sol_f are Matrix types of size
+  # (Number of face plotting nodes) x (Number of faces).
+  function face_first_reshape(x, num_nodes_1D, num_nodes, num_elements)
+      num_reference_faces = 2 * ndims(mesh)
+      xf = view(reshape(x, num_nodes, num_elements), vec(Fmask), :)
+      return reshape(xf, num_nodes_1D, num_elements * num_reference_faces)
+  end
+  reshape_and_interpolate(x) = plotting_interp_matrix1D * face_first_reshape(x, nnodes(dg), n_nodes_2d, n_elements)
+  xfp, yfp, ufp = map(reshape_and_interpolate, (x, y, u.data))
+
+  return xfp, yfp, ufp
+end
+
+function mesh_plotting_wireframe(u::uType, mesh, equations, dg::DGMulti, cache;
+                                 nvisnodes=2*nnodes(dg)) where {uType <: ScalarData}
```

Comment:
Why not
```suggestion
function mesh_plotting_wireframe(u::ScalarData, mesh, equations, dg::DGMulti, cache; nvisnodes=2*nnodes(dg))
```
like above?